src/app/components/myProfile/myProfile.controller.js

```diff
@@ -264,19 +258,16 @@
       }
 
       function updateKits() {
-        var kitIDs = _.map(vm.user.kits, 'id');
-        if(!kitIDs.length) {
+        if(!vm.user.kits.length) {
           return [];
         }
 
-        $q.all(
-          kitIDs.map(function(id) {
-            return device.getDevice(id)
-              .then(function(data) {
-                return new PreviewKit(data);
-              });
+        device.createKitBlueprints().then(function(){
```

Comment:
why a second time? it has been called in the initializer. Again, for me better in the `resolve` of the "profile" route.